lib/mllp/receiver.ex

```diff
@@ -328,4 +343,30 @@ defmodule MLLP.Receiver do
       framing_context2
     end
   end
+
+  defp normalize_ip({_, _, _, _} = ip), do: ip
```

Comment:
One suggestion :

I would suggest the latest version of this function I had in this WIP : https://github.com/HCA-Healthcare/elixir-mllp/pull/27/files#diff-4fc1cf0a1c259a35f3a58eb89534250fbbad4a234f6845390f94d916220a6343R79

I'm not sure about getaddr/2 call below. That'll perform a dns lookup  and in the case the ip is an ipv4 or ipv6 as a binary it will fail  with `{:error, :einval}` 


To validate an ip address in the case that it's a binary is tricky. You can run `:inet.parse*` but if it's  a host name it'll fail. 

You could try one if it fails, then try the other, if both fail we know it's one of two things : 

1. The ip address is bad, possibly the hostname is bad
2. The hostname might be fine,but name resolution failed. 

Is your intent that name resolution must succeed anyway?

Reply:
Glad you raised this point. I glanced through methods of `inet` and two close ones were `getaddr/2` and `parse_address` for parsing binary IP address. 
I noticed, `getaddr/2` could also parse 'localhost', '127.0.0.1' and {127, 0, 0, 1} but `parse_address` could not as it only worked with binary. But then `getaddr` also takes `address_family` as parameter which is basically if it's IPV4 vs IPV6. I chose IPV4 only because I felt there are not many client with MLLP protocol use IPV6 (I may be wrong).

Both `parse_address` and `getaddr` don't do dns lookup, whatever DNS I tried it failed except for `localhost`. And I thought we really don't need to use DNS and just use IP. 
 
After thinking through this more though, this is what we should do:

- use `parse_address` for binary IP
- if parse_address fails, try use `gethostbyname/1` (but it also returns a list of IPs, not sure why and in which case. DNS I tried, I got one IP in list)
- if both fails, then its invalid IP or DNS

thoughts?

Reply:
Ahh yeah, you are right, it will work with an ip address. The folly of going back and forth betwen elixir and erlang is I'll see `"1.1.1.1"` and sometimes my brain says "binary!" and then sometimes it says "string" (char list) :p 

Yeah, I think your proposal is sound so long as dns failure should trigger the error so to speak. Since this will only happen with and  `allowed_ips` option and not effect the general case, it should be fine :) We might find ourselves coming  back to make an option for this for this option later, but that's fine too :) Kaizen!

Reply:
Ahh but not, it does indeed do  dnslookup ... 

```
:inet.getaddr('erlef.org', :inet)
{:ok, {66, 42, 87, 202}}
``` 

```
nslookup erlef.org
...

Non-authoritative answer:
Name:	erlef.org
Address: 66.42.87.202
```

Reply:
Interesting, yup `getaddr` does work for dns lookup. I may have missed something when trying. It looks like it is using `gethostname` method internally as well. In order to add IPV6 support though we would be better off using `parse_address/1` and `gethostbyname/1` method as suggested above. It will cover both IPV4 and IPV6.

If DNS lookup, or parse address fails, we can filter out failed address from the list of allowed IPs with a warning Or raise an error. I am currently filtering with a warning

Reply:
Since the option is `allowed_ips` how about we simply raise (or warn) if it's not an  ip address vs trying to do a DNS lookup?

Reply:
I can change `allowed_ips` to `allowed_servers`

Reply:
That'll work too 👍